### DISCUSSION TITLE
Add an action to the header over the actions column

### DISCUSSION THREAD

**Discussion by adamkelso** [awcodes/filament-table-repeater#122] [Ideas]
Not sure if anyone else would appreciate this, but could it make sense to allow an action to be placed in the header above the actions column? Meaning, instead of showing the default Add action below the table, I'd like to set a Plus icon button in the header above the actions column. This screenshot might illustrate this idea a little better than words.

### Standard Table
<img width="862" alt="Screenshot 2024-06-24 at 2 24 48 PM" src="https://github.com/awcodes/filament-table-repeater/assets/2157765/8b85e559-37ee-4012-9d93-eec84b980025">

### Streamlined Table
<img width="865" alt="Screenshot 2024-06-24 at 2 25 40 PM" src="https://github.com/awcodes/filament-table-repeater/assets/2157765/63c84c47-5e33-4335-b4d1-43976bfbea8a">

If that's something the community would appreciate, I'd be more than happy to create a PR. If not, no hard feelings.

I'm not entirely sure what a good method name for this would be. Maybe `actionsColumnHeaderAction`?

**Comment by awcodes** [comment]
I'm not opposed to it. 2 things to consider though. 

1. What would this be like from an a11y standpoint since the header would no longer read as 'actions' for that column?

2. It's on the v4 roadmap to bring table repeater into Filament core, so when that happens this plugin will most likely get deprecated shortly after that release. So the functionality can't be guaranteed with the core integration.

**Reply by adamkelso** [reply]
Awesome!

1. For the a11y concern, the label is currently surrounded by a `span` tag that is visible to screen readers only. I could either leave that HTML untouched and make the new action a sibling of the `span`, or we could make the label an attribute of the new action. I'll do a little research, but I think just making it a sibling should still be entirely compliant.
2. So glad table will be added to the core. And understood that there is a risk of it disappearing in V4.

**Reply by awcodes** [reply]
For the api, I think something like columnHeaderActions() to keep in line with extraItemActions() there is already a way to disable the add action in general, so maybe leverage that as a way to 'move' it to the header.

**Reply by awcodes** [reply]
Could possibly just be headerActions() too.

**Reply by awcodes** [reply]
Would have to check the underlying repeater to see if there are any namespace collisions for the method names.
